### ISSUE TITLE
Remove `AppStore.getBranchesState`

### ISSUE THREAD

**Issue by iAmWillShepherd** [desktop/desktop#7231]
I discovered this after noticing that `AppStore._createStash` function did not behave as expected. After doing a bit of digging, it turns out that my test repo not being a `GitHubRepository` was the cause. I confirmed with @desktop/engineering that getting `BranchesState` does not depend on `GitHubRepository`. 

**Problem function**
https://github.com/desktop/desktop/blob/92faa65e495a38c2be373fac16f8b4ea74a7bcac/app/src/lib/stores/app-store.ts#L1621-L1628

This function should be removed because of the following reasons:
* `repository` is optional
* `undefined` is returned if the `repository.githubRepository` is null
* All functions that use this function are circumventing explicit state access

Basically this function does not follow the patterns we've defined elsewhere.
